sw/device/tests/BUILD

```diff
@@ -1622,6 +1590,219 @@ opentitan_test(
     ],
 )
 
+cc_library(
+    name = "pwrmgr_sleep_resets_lib",
+    srcs = ["pwrmgr_sleep_resets_lib.c"],
+    hdrs = ["pwrmgr_sleep_resets_lib.h"],
+    target_compatible_with = [OPENTITAN_CPU],
+    visibility = ["//sw/device/tests:__pkg__"],
+    deps = [
+        "//hw/top_earlgrey/sw/autogen:top_earlgrey",
+        "//sw/device/lib/base:abs_mmio",
+        "//sw/device/lib/base:mmio",
+        "//sw/device/lib/dif:alert_handler",
+        "//sw/device/lib/dif:aon_timer",
+        "//sw/device/lib/dif:flash_ctrl",
+        "//sw/device/lib/dif:pinmux",
+        "//sw/device/lib/dif:pwrmgr",
+        "//sw/device/lib/dif:rstmgr",
+        "//sw/device/lib/dif:rv_plic",
+        "//sw/device/lib/dif:rv_timer",
+        "//sw/device/lib/dif:sysrst_ctrl",
+        "//sw/device/lib/runtime:irq",
+        "//sw/device/lib/runtime:log",
+        "//sw/device/lib/testing:alert_handler_testutils",
+        "//sw/device/lib/testing:aon_timer_testutils",
+        "//sw/device/lib/testing:pinmux_testutils",
+        "//sw/device/lib/testing:pwrmgr_testutils",
+        "//sw/device/lib/testing:rstmgr_testutils",
+        "//sw/device/lib/testing:rv_plic_testutils",
+        "//sw/device/lib/testing/test_framework:check",
+    ],
+)
+
+opentitan_test(
+    name = "pwrmgr_random_sleep_all_reset_reqs_test",
+    srcs = ["pwrmgr_random_sleep_all_reset_reqs_test.c"],
+    cw310 = new_cw310_params(
+        test_cmd = """
```

Comment:
Can you explain why you put line 1629,1630?

Reply:
This seems needed by the BUILD rules: if we don't add some test_cmd the default command causes errors. You'll notice all tests that set a test_harness also add a test_cmd. I hope this will be fixed.

Reply:
the default test_cmd assumes opentitantool, if you specify a test_harness, you almost always need to override the test_cmd since opentitantool and opentitanlib-based tests don't use the same command line arguments